src/components/Synthetics/TradeHistoryRow/helpers.ts

```diff
@@ -0,0 +1,382 @@
+
+type FormatPositionMessageChunk = {
+  text: string;
+  textRed?: boolean;
+  tooltipRows?: StatsTooltipRowProps[];
+  tooltipTitle?: string;
+  tooltipTitleRed?: boolean;
+  tooltipFooterRed?: boolean;
+  tooltipFooter?: string;
+};
+type FormatPositionMessageResult = FormatPositionMessageChunk[] | null;
+
+export function getOrderActionText(tradeAction: TradeAction) {
+  let actionText = "";
+
+  if (tradeAction.eventName === TradeActionType.OrderCreated) {
+    actionText = t`Create`;
+  }
+
+  if (tradeAction.eventName === TradeActionType.OrderCancelled) {
+    actionText = t`Cancel`;
+  }
+
+  if (tradeAction.eventName === TradeActionType.OrderExecuted) {
+    actionText = t`Execute`;
+  }
+
+  if (tradeAction.eventName === TradeActionType.OrderUpdated) {
+    actionText = t`Update`;
+  }
+
+  if (tradeAction.eventName === TradeActionType.OrderFrozen) {
+    actionText = t`Freeze`;
+  }
+
+  return actionText;
+}
+
+export const formatPositionMessage = (
+  tradeAction: PositionTradeAction,
+  minCollateralUsd: BigNumber
+): FormatPositionMessageResult => {
+  const indexToken = tradeAction.indexToken;
+  const priceDecimals = tradeAction.indexToken.priceDecimals;
+  const collateralToken = tradeAction.initialCollateralToken;
+  const sizeDeltaUsd = tradeAction.sizeDeltaUsd;
+  const collateralDeltaAmount = tradeAction.initialCollateralDeltaAmount;
+
+  const isIncrease = isIncreaseOrderType(tradeAction.orderType);
+  const increaseText = isIncrease ? t`Increase` : t`Decrease`;
+  const longText = tradeAction.isLong ? t`Long` : t`Short`;
+  const positionText = `${longText} ${indexToken.symbol}`;
+  const sizeDeltaText = `${isIncrease ? "+" : "-"}${formatUsd(sizeDeltaUsd)}`;
+
+  switch (tradeAction.orderType) {
+    case OrderType.LimitIncrease:
+    case OrderType.LimitSwap:
+    case OrderType.LimitDecrease:
+    case OrderType.StopLossDecrease: {
+      const triggerPrice = tradeAction.triggerPrice;
+      const executionPrice = tradeAction.executionPrice;
+      const pricePrefix = getTriggerThresholdType(tradeAction.orderType!, tradeAction.isLong!);
+      const actionText = getOrderActionText(tradeAction);
+      const tokenPrice = getTokenPriceByTradeAction(tradeAction);
+
+      if (tradeAction.eventName === TradeActionType.OrderExecuted) {
+        const executeOrderStr = t`Execute Order: ${increaseText} ${positionText} ${sizeDeltaText},`;
+        return [
+          {
+            text: `${executeOrderStr} `,
+          },
+          {
+            text: [
+              tokenPrice ? t`Triggered at: ${formatUsd(tokenPrice, { displayDecimals: priceDecimals })}` : undefined,
+              t`Execution Price: ${formatUsd(executionPrice, { displayDecimals: priceDecimals })}`,
+            ]
+              .filter(Boolean)
+              .join(", "),
+            tooltipRows: getExecutionPriceTooltipRows(tradeAction),
+          },
+        ];
+      }
+
+      const isFailed = tradeAction.eventName === TradeActionType.OrderFrozen;
```

Comment:
probably it's better to keep isFrozen naming to avoid confusion with cancelled orders